### DISCUSSION TITLE
Dealing with large schemas and static type checking

### DISCUSSION THREAD

**Discussion by bellybuttonbackwards** [RobertCraigie/prisma-client-py#867] [Q&A] [answered]
Hello,

We have a reasonably large `schema.prisma` file, currently at 1859 lines, and we've just crossed the threshold where the `prisma-client-py` generated `types.py` file is 51MB. This is causing our pyright static type checking to fail because pyright has [a file size limit of _50MB_ ](https://github.com/microsoft/pyright/blob/c88987e5ac8afa63ebf392c469e4e538e19a02d0/packages/pyright-internal/src/analyzer/sourceFile.ts#L459-L465). Specifically, the checking we do in CI to verify whether a commit is safe to merge is now failing with many `reportUnknownArgumentType`/`reportUnknownMemberType`/etc errors that were previously valid when the `types.py` file was small enough to analyze.

I'm surprised I wasn't able to find any existing issues referencing this sort of thing before, does the community have suggestions for how to configure our schema or the generator to generate smaller files?

**Comment by RobertCraigie** [comment, marked as answer]
Sorry about this, have you enabled recursive types? https://prisma-client-py.readthedocs.io/en/stable/reference/config/#recursive

**Reply by bellybuttonbackwards** [reply]
sorry, I should have included my generator block. We have
```
generator clientpy {
  provider             = "prisma-client-py"
  interface            = "asyncio"
  recursive_type_depth = 5
  previewFeatures      = ["postgresqlExtensions"]
  output               = "../../../pipelines/pipelines/prisma"
}
```
but reading the docs you linked in more detail, since we _are_ using pyright, i set the value to `recursive_type_depth = -1` and that dropped the generated size to `11MB` which seems like it gives us plenty of headroom.

Thanks so much for the quick reply!

matt
